internal/pkg/service/common/etcdclient/etcdclient.go

```diff
@@ -200,10 +196,10 @@ func New(ctx context.Context, proc *servicectx.Process, tracer trace.Tracer, end
 		c.KV = etcdlogger.KVLogWrapper(c.KV, logger.DebugWriter())
 	}
 
-	// Sync endpoints list from cluster, it is used also as a connection check.
-	if err := c.Sync(connectCtx); err != nil {
+	// Connection check: get cluster members
+	if _, err := c.MemberList(connectCtx); err != nil {
 		_ = c.Close()
-		return nil, errors.Errorf("cannot create etcd client: cannot sync cluster members: %w", err)
+		return nil, errors.Errorf("cannot create etcd client: cannot get cluster members: %w", err)
```

Comment:
V `etcdclient` sme na zaciatku volali `Sync`, to nacita `MembersList` a ulozi ho do `endpoints`.
- Tj. namiesto nami zadaneho endpointu, sa kontaktuju priamo jednotlive `etcd nodes` a samotny etcd client si robi load-balancing.
- To ale nijak nedetekuje `unhealthy` nodes, preto to vypinam, a budeme mat iba jeden endpoint `abc`, load balancing na `nodes` riesi `k8s`, ... `unehealthy` nodes su automaticky odpojene z `k8s` service.